ysh/funcs_builtin.py

```diff
@@ -192,6 +195,39 @@ def Init3(mem, config_parser, eval_to_dict, block_as_str, hay_result):
     SetGlobalFunc(mem, 'block_as_str', block_as_str.Call)
 
 
+class _Bool(vm._Func):
+
+    def __init__(self):
+        # type: () -> None
+        vm._Func.__init__(self)
+
+    def Run(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+        assert len(pos_args) == 1
+        assert len(named_args) == 0
+        return value.Bool(expr_eval.ToBool(pos_args[0]))
+
+
+class _StrStartsWith(vm._Func):
+
+    def __init__(self):
+        # type: () -> None
+        vm._Func.__init__(self)
+
+    def Run(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+        assert len(pos_args) == 2
+        assert len(named_args) == 0
+
+        UP_s = pos_args[0]
+        UP_prefix = pos_args[1]
+        assert UP_s.tag() == value_e.Str
+        assert UP_prefix.tag() == value_e.Str
+        s = cast(value.Str, UP_s)
+        prefix = cast(value.Str, UP_prefix)
+        return value.Bool(s.s[:len(prefix.s)] == prefix.s)
```

Comment:
actually I realized here we could just use `s.startswith(prefix.s)`

that will translate fine, since our `Str` implementation has it

Reply:
ah right. good catch